backend-tests/tests/test_deployments.py

```diff
@@ -0,0 +1,494 @@
+                    {
+                        # Start time is optional in the first stage (default to now)
+                        "batch_size": 10,
+                    },
+                    {
+                        "start_ts":
+                        str((datetime.utcnow() +
+                             timedelta(days=1)).isoformat("T")) + "Z",
+                        "batch_size":
+                        10,
+                    },
+                    {
+                        "start_ts":
+                        str((datetime.utcnow() +
+                             timedelta(days=1)).isoformat("T")) + "Z",
+                        "batch_size":
+                        10,
+                    },
+                    {
+                        "start_ts":
+                        str((datetime.utcnow() +
+                             timedelta(days=1)).isoformat("T")) + "Z",
+                        "batch_size":
+                        10,
+                    },
+                    {
+                        "start_ts":
+                        str((datetime.utcnow() +
+                             timedelta(days=2)).isoformat("T")) + "Z",
+                        # Batch size is optional in the last stage (ie, it is the remaining devices)
+                    },
+                ],
+            },
+            {
+                "name":
+                "Five phases, first no start time, last start time, no batch size",
+                "artifact_name":
+                "deployments-phase-testing",
+                "device_count":
+                100,
+                "phases": [{
+                    "batch_size": 10,
+                    "device_count": 0,
+                }, {
+                    "batch_size": 10,
+                    "device_count": 0,
+                }, {
+                    "batch_size": 10,
+                    "device_count": 0,
+                }, {
+                    "batch_size": 10,
+                    "device_count": 0,
+                }, {
+                    "batch_size": 60,
+                    "device_count": 0,
+                }],
+            }),
+    ]
+
+    @pytest.mark.parametrize("test_case", test_cases)
+    def test_phased_deployments_success(self, test_case,
+                                        setup_deployments_enterprise_test):
+
+        user, tenant, utoken = setup_deployments_enterprise_test
+        request_body, expected_response = test_case
+        deploymentclient = ApiClient(
+            api.client.GATEWAY_URL + "/api/management/v1/deployments")
+        resp = deploymentclient.with_auth(utoken).call(
+            'POST',
+            '/deployments',
+            body=request_body,
+        )
+        assert resp.status_code == 201
+        deployment_id = os.path.basename(resp.headers['Location'])
+        resp = deploymentclient.with_auth(utoken).call('GET', '/deployments')
+        assert resp.status_code == 200
+        assert len(resp.json()) == 4
```

Comment:
just a note - it would be useful to actually compare the existing deployments to those returned - so that we're 100% sure they're unchanged.

but, I think we can cut a corner and leave it like this. it would be good to keep in mind with next test tasks.